CppCoreGuidelines.md

```diff
@@ -4701,7 +4701,7 @@ Users will be surprised if copy/move construction and copy/move assignment do lo
         };
         shared_ptr<Impl> p;
     public:
-        Silly(const Silly& a) : p{a.p} { *p = *a.p; }   // deep copy
+        Silly(const Silly& a) : p(make_shared<Impl>()) { *p = *a.p; }   // deep copy
```

Comment:
Why not `p{make_shared<Impl>(*a.p)}` instead?

Reply:
its trying to be symmetric with the assignment to reveal the difference, update both if you do this,